src/process_utility.c

```diff
@@ -1433,6 +1669,11 @@ process_cluster_start(Node *parsetree, ProcessUtilityContext context)
 			ts_chunk_index_mark_clustered(cim->chunkoid, cim->indexoid);
 
 			/* Do the job. */
+
+			/*
+			 * Since we keep OIDs between transactions, there is a potential
```

Comment:
I don't understand this. How can OIDs be reassigned between txns?

Reply:
OID are assigned ~sequentially, if we get enough of them they'll wrap around, and start referring to new objects.

Reply:
The locking idea (session locks on chunks that are released when each index is created) would prevent this as well.